pkg/election/election.go

```diff
@@ -320,15 +320,16 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
 			continue
 		}
 
+		// make sure `elec.Campaign` was exited (instead of rely on `<-eleObserveCh`), to avoid data race writing `elec`'s member
+		campaignWg.Wait()
+		cancel2()
```

Comment:
Can we move it in front of L309 now?

Reply:
I guess can't. For multi-leader election, non-leader's `elec.Campaign` will block thus can't `campaignWg.Done()`. If we move `campaignWg.Wait()` before above two if, behaviour will change to "failed campaign won't restart a new one". Not sure this is expected